auto-setup.Dockerfile

```diff
@@ -1,26 +0,0 @@
-ARG GOPROXY
```

Comment:
Long time ago we had one `Dockerfile`. Then I split it "per image". There was a good reason for it, but I couldn't recall. Why do you merging them back?

Reply:
> I combined the auto-setup and server dockerfiles to work around an issue with (as far as I can tell) docker buildx bake and ENTRYPOINTs in multi-stage builds.

Buildx won't let you override the entrypoint the way we needed to in a multi-stage build. I have no idea why, but it doesn't let you do it.